sql/handler.cc

```diff
@@ -6018,7 +6012,7 @@ bool ha_show_status(THD *thd, handlerton *db_type, enum ha_stat_type stat)
   }
   else
   {
-    if (db_type->state != SHOW_OPTION_YES)
+    if (!ha_storage_engine_is_enabled(db_type))
```

Comment:
We can't get `binlog` here as it is `HTON_NOT_USER_SELECTABLE`. So just remove `hton->state` check.